rfcs/011-dedicated-renku-git-user/011-dedicated-renku-git-user.md

```diff
@@ -0,0 +1,113 @@
+In addition with the switch from Gitlab user access tokens to project tokens there was a
+lot of automatically generated "bot" users that Gitlab created when the project tokens
+were created. This is just what happens when a Gitlab project token is created but it may
+be cause concerns with some admins or users.
+
+Lastly, with having a regular user being dedicated to the KG, there is no more impersonation
+of users and "fake" user activity that comes from the periodic checks and indexing of projects
+that is done by the KG. All this activity will now be associated with the Renkubot user rather
+than individual real users.
+
+## Design Detail
+
+### Helm Chart
+
+In the Helm chart we would add a hook that would create a user in the Gitlab deployment
+that Renku is connected to. The username and password will be saved in a secret. The KG
+would then have access to the secret and use it to authenticate with Gitlab.
+
+Alternatively the Helm chart can also accept existing user credentials. 
+
+### Gateway
+
+Either as part of this RFC or as an improvement the management of this secret / tokens for 
+the Renkubot user should be done by the Gateway and the KG will simply either ask the Gateway 
+for credentials or a Gateway's forward proxy will inject the right credentials in the right calls.
+
+This will also be helpful when we support multiple Gitlabs/Github. In this case the KG does
+not need to keep track which credentials to use for which Gitlab. The gateway can figure this
+out and inject the right credentials for the Renkubot user.
+
+### UI
+
+There are some changes needed in the UI for this. Namely:
+- To get a list of all projects for a user the UI would have to call Gitlab on behalf
+of the user. This is because the KG will now only know about projects to which the Renkubot
+user has been added. In other words, the KG will not know about any projects for which 
+KG integration is not active. It could figure this out for public Gitlab projects
+but it would be impossible to find private projects that are not in the KG.
+- Creating a project on Renku will have to include adding this Renkubot user to the project
+with `Reporter` permissions.
+- KG integration for a specific project can still be verified with the KG. In some cases
+the KG will simply not know anything at all for a project. In this case KG integration
+for this project is not enabled. Checking that the Renkubot user is in the project with the
+right permissions is not enough. It could be that the project has the Renkubot user but
+the KG has not fully finished indexing the project.
+- Enabling KG integration on projects that do not have it will have to be modified to also
+add the user to the project similarly to creating a project.
+- Disabling / removing KG integration results in removal of the project from the KG and removing
+the Renkubot user from the project.
+
+### KG
+
+Depending on when the Gateway refactoring is completed, the KG may not have to deal with the Renkubot
```

Comment:
I guess it's not a big difference for the KG which option we'd choose. At least from the implementation side, I think it might be even easier if we'd get the credentials injected so we can simply exchange them for an OAuth token in the token repository.